include/seqan3/argument_parser/detail/format_parse.hpp

```diff
@@ -66,7 +66,7 @@ class format_parse : public format_base
      * \param[in] argc_ The number of command line arguments.
      * \param[in] argv_ The command line arguments to parse.
      */
-    format_parse(int const argc_, std::vector<std::string> && argv_) :
+    format_parse(int const argc_, std::vector<std::string> argv_) :
```

Comment:
Why copy?

Reply:
this change, in commit fe5ec6628f7cb34f19aa87a24809a26c08d3de71, is part of storing the command line arguments in the argument parser (before I did not need them and forwarded them to `format_parse` but now I need them to enable searching if an option was set).
Since I store them in a variable, I don't pass a forwarded rvalue to `format_parse` anymore (see [argument_parser.hpp:709](https://github.com/seqan/seqan3/pull/1859/commits/fe5ec6628f7cb34f19aa87a24809a26c08d3de71#diff-18eb4ceee3f51232eaef3efff94f6f15R709)) but instead I pass an lvalue. Since format_parse is moving `argv_` into its member `argv` in the member initialisation below, I need a copy here.

Reply:
Here an explanation why you should prefer copy and move over const-ref and copy. The advantage is that rvalues will never be copied but only moved (not if you capture them by const-ref and then copy inside the function). lvalues need to be copied at least once anyway and from the performance perspective it doesn't matter if it happens inside of the function or in the function parameter. One additional advantage of copying in the function parameter is that it is easier to write a strong exception safe function since the body of the function is not executed if something throws during that copy.